pyo3-macros-backend/src/pyclass.rs

```diff
@@ -681,14 +961,155 @@ fn enum_default_methods<'a>(
+            #deprecations
+            ::std::result::Result::Ok(py.get_type::<#variant_cls>().into())
+        }
+    };
+
+    let method_def = quote! {
+        _pyo3::class::PyMethodDefType::ClassAttribute({
+            _pyo3::class::PyClassAttributeDef::new(
+                #python_name,
+                _pyo3::impl_::pymethods::PyClassAttributeFactory(#cls_type::#wrapper_ident)
+            )
+        })
+    };
+
+    MethodAndMethodDef {
+        associated_method,
+        method_def,
+    }
+}
+
+fn complex_enum_variant_new<'a>(
+    cls: &'a syn::Ident,
+    variant: &'a PyClassComplexEnumVariant<'a>,
+) -> Result<MethodAndSlotDef> {
+    let variant_cls = format_ident!("{}_{}", cls, variant.ident);
+    let variant_cls_type: syn::Type = parse_quote!(#variant_cls);
+
+    let arg_py_ident: syn::Ident = parse_quote!(py);
+    let arg_py_type: syn::Type = parse_quote!(_pyo3::Python<'_>);
+
+    let args = {
+        let mut no_pyo3_attrs = vec![];
+        let attrs = crate::pyfunction::PyFunctionArgPyO3Attributes::from_attrs(&mut no_pyo3_attrs)?;
+
+        let mut args = vec![
+            // py: Python<'_>
+            FnArg {
+                name: &arg_py_ident,
+                ty: &arg_py_type,
+                optional: None,
+                default: None,
+                py: true,
+                attrs: attrs.clone(),
+                is_varargs: false,
+                is_kwargs: false,
+            },
+        ];
+
+        for field in &variant.fields {
+            args.push(FnArg {
+                name: field.ident,
+                ty: field.ty,
+                optional: None,
+                default: None,
+                py: false,
+                attrs: attrs.clone(),
+                is_varargs: false,
+                is_kwargs: false,
+            });
+        }
+        args
+    };
+    let signature = crate::pyfunction::FunctionSignature::from_arguments(args)?;
+
+    let spec = FnSpec {
+        tp: crate::method::FnType::FnNew,
+        name: &format_ident!("__pymethod_constructor__"),
+        python_name: format_ident!("__new__"),
+        signature,
+        output: variant_cls_type.clone(),
+        convention: crate::method::CallingConvention::TpNew,
+        text_signature: None,
+        unsafety: None,
+        deprecations: Deprecations::default(),
     };
-    let options = EnumVariantPyO3Options::take_pyo3_options(&mut variant.attrs)?;
-    Ok(PyClassEnumVariant { ident, options })
+
+    crate::pymethod::impl_py_method_def_new(&variant_cls_type, &spec)
+}
+
+fn complex_enum_variant_field_getter<'a>(
+    variant_cls_type: &'a syn::Type,
+    field_name: &'a syn::Ident,
+    field_type: &'a syn::Type,
+    field_span: Span,
+) -> Result<MethodAndMethodDef> {
+    let signature = crate::pyfunction::FunctionSignature::from_arguments(vec![])?;
+
+    let self_type = crate::method::SelfType::Receiver {
+        mutable: false,
+        span: field_span,
+    };
```

Comment:
I think you need to change this to `SelfType::TryFromPyCell` to make the field getters compile. (This is the cause of your generated getter trying to take `&self`.)

Reply:
This helped, thanks! The fix is a little ugly, but it works now! :)